src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,364 @@
+package org.openstreetmap.atlas.geography.atlas.builder.proto;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.Latitude;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Longitude;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.AtlasMetaData;
+import org.openstreetmap.atlas.geography.atlas.builder.AtlasSize;
+import org.openstreetmap.atlas.geography.atlas.builder.RelationBean;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.ItemType;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlasBuilder;
+import org.openstreetmap.atlas.geography.atlas.pbf.slicing.identifier.ReverseIdentifierFactory;
+import org.openstreetmap.atlas.geography.converters.proto.ProtoLocationConverter;
+import org.openstreetmap.atlas.geography.converters.proto.ProtoTagListConverter;
+import org.openstreetmap.atlas.proto.ProtoArea;
+import org.openstreetmap.atlas.proto.ProtoAtlasContainer;
+import org.openstreetmap.atlas.proto.ProtoAtlasContainer.Builder;
+import org.openstreetmap.atlas.proto.ProtoEdge;
+import org.openstreetmap.atlas.proto.ProtoLine;
+import org.openstreetmap.atlas.proto.ProtoLocation;
+import org.openstreetmap.atlas.proto.ProtoNode;
+import org.openstreetmap.atlas.proto.ProtoPoint;
+import org.openstreetmap.atlas.proto.ProtoRelation;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.streaming.resource.WritableResource;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * Build an Atlas from a naive proto serialized file, or write an Atlas to a naive proto serialized
+ * file. At this stage, the builder can only handle Atlases that have tagged Points and Lines.
+ *
+ * @author lcram
+ */
+public class ProtoAtlasBuilder
+{
+    private final ProtoLocationConverter protoLocationConverter = new ProtoLocationConverter();
```

Comment:
One nit: Why not make all the converters `private static final` variables?

Reply:
Haha. I had this initially, but it looked more verbose with the all-caps and the fact that the style checker prepends ProtoAtlasBuilder. to the front of every variable reference. I can change it back to static though if that is preferred.